### PR TITLE
Improve installation and support instructions

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #2909

### CODE REVIEW

**Review by ajinkyau — APPROVED** [review]

**Comment by konstruktoid** [comment]
LGTM, but https://github.com/ansible/ansible-lint/issues/2909#issuecomment-1396864703 still occurs.
Most likely an edge case for some reason, but fyi.

File: docs/installing.md [hunk 1]
```diff
@@ -6,21 +6,48 @@
...
+Ansible-lint team does not support any installation methods that are not
+mentioned in this document. Before raising any bugs related to installation,
+ensure that you have followed the instructions below:
+
+- The installation method is mentioned in this document
+- You already upgraded the installer (`pip` or `pipx`) to the latest version
+  available from pypi.org. If you installed them using your system package, you
+  will need to install newer versions.
+- Installing from git zip archives is not supported but it should work at least
+  with main branch and latest versions of pip and setuptools.
```
Comment (oraNod) [inline docs/installing.md:26]:
```suggestion
- If you are installing from a git zip archive, which is not supported but should work,
  ensure you use the main branch and the latest version of pip and setuptools.
```

File: docs/installing.md [hunk 2]
```diff
@@ -6,21 +6,48 @@
...
+- If you are using a the tool from within a container or system package, you
+  should not report the issue here. Please contact the packager for that.
+- If you are using [poetry](https://python-poetry.org/), read
+  [discussion](https://github.com/ansible/ansible-lint/discussions/2820#discussioncomment-4400380)
+
+Pull requests to improve installation instructions are welcome, but please do
+not open new issues as these will be closed and locked.
```
Comment (oraNod) [inline docs/installing.md:33]:
```suggestion
Pull requests to improve installation instructions are welcome. Any new issues related to
installation will be closed and locked.
```

**Review by oraNod — APPROVED** [review]
LGTM

**Pull request merged**